### ISSUE TITLE
Global DNS installation fails to install on rancher with local k8s 1.22.4-rancher1-1

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#35909]
Information about the Cluster
**Rancher Server Setup**
- Rancher version: v2.6-head ed77b51
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1, v1.22.4-rancher1-1, 1.3.3-rc6
- Proxy/Cert Details: Selfsigned

**Information about the Cluster**
- Kubernetes version: v1.22.4-rancher1-1
- Cluster Type (Local/Downstream): `Local 3  worker + etcd + cp RKE1`

**Describe the bug**
Installing global DNS from ember UI by navigating to url/g fails with an error. The workload pod `systemapp-anu-dns-rancher-external-dns` is not deployed and the app is stuck in `installing`

**To Reproduce**
1. Create a rancher server on local k8s 1.22.4-rancher1-1 on an RKE1 cluster with 3 worker + cp + etcd
2. From URL/g --> Tools --> Global DNS providers --> provide the following details and create
```
name: test-dns
domain, AWS_Access, aws_secret
```
**Result**
The installation fails with the error : 
```
Failed to install app systemapp-anu-dns. Error: validation failed: [unable to recognize "": no matches for kind "ClusterRole" in version "rbac.authorization.k8s.io/v1beta1", unable to recognize "": no matches for kind "ClusterRoleBinding" in version "rbac.authorization.k8s.io/v1beta1"]

```

**Expected Result**
Expected the installation to go through without any errors.

**Additional context**
Installation goes through if the local cluster is on version `v1.21`

**Comment by prachidamle** [comment]
`no matches for kind "ClusterRole" in version "rbac.authorization.k8s.io/v1beta1", unable to recognize "": no matches for kind "ClusterRoleBinding" in version "rbac.authorization.k8s.io/v1beta1"`

Seems this system-chart needs to be updated to use ClusterRole/v1 and not ClusterRole/v1beta1. Same update  ClusterRoleBinding/v1beta1 to ClusterRoleBinding/v1

**Comment by MKlimuszka** [comment]
This has been worked on, and is likely fixed. Sending to QA for validation with a low priority.

**Comment by AndrewHoffmanQA** [comment]
per conversation with @ronhorton this is a team 2 bug.

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|As an admin user, create a Global DNS provider and validate that it is able to be created successfully|

**Comment by markusewalker** [comment]
Validated this is addressed in `v2.6-head` https://github.com/rancher/rancher/commit/4485a0d6aaef024bb54e700d073394034b638a98.

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.6-head` https://github.com/rancher/rancher/commit/4485a0d6aaef024bb54e700d073394034b638a98
- RKE version: v1.3.19
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As an admin user, create a Global DNS provider and validate that it is able to be created successfully|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Setup Rancher with local K8s cluster using RKE1 v1.24.10-rancher1-1 and RKE v1.3.19.
2. Created a Global DNS provider with my root domain and AWS credentials.
3. Created a Global DNS entry using the global DNS provider that I previously created.
4. On the local cluster, deployed a test ingress with my Rancher FQDN and annotation with key:rancher.io/Global DNS entry, value: Rancher FQDN.
5. Validated that the Ingress was created successfully and came up `Active`.
